dias/analyzers/dataset_analyzer.py

```diff
@@ -200,11 +201,30 @@ def run(self):
                     ),
                 )
 
+                self.validate_null(_file, ad)
                 self.validate_flag_updates(_file, ad, flg)
                 self.validate_freqs(_file, ad)
 
                 self.register_done([_file])
 
+    def validate_null(self, filename, ad):
+        """
+        Check that data that is considered valid (i.e. frac_lost < 1) does not have a null dataset ID.
+
+        Paramaters
+        ----------
+        filename : String
+            path to file loaded in ad
+        ad : andata.CorrData
+        """
+        file_ds = ad.flags["dataset_id"][:] == "00000000000000000000000000000000"
```

Comment:
Have you tested this code? I usually fine these need to be byte strings, but maybe it works here as a standard str.
```suggestion
        file_ds = ad.flags["dataset_id"][:] == b"00000000000000000000000000000000"
```

Reply:
It worked as a standard string! But if the byte string also works for me, I will use it.

Reply:
The byte string comparison did not work for me, and I am noticing that everywhere else in dataset analyzer we are using the standard string
so I am going to keep it as a standard string
This happens:
>>> file_ds = ad.flags["dataset_id"][:] == b"00000000000000000000000000000000"
>>> file_ds
False
>>> file_ds = ad.flags["dataset_id"][:] == "00000000000000000000000000000000"
>>> file_ds
array([[False, False, False, ..., False, False, False],
       [ True,  True,  True, ...,  True,  True,  True],
       [ True,  True,  True, ...,  True,  True,  True],
       ...,
       [ True,  True,  True, ...,  True,  True,  True],
       [ True,  True,  True, ...,  True,  True,  True],
       [ True,  True,  True, ...,  True,  True,  True]])
I think it interprets the object on the left as bytes, instead of comparing each element